pkg/provider/azure.go

```diff
@@ -1318,6 +1332,32 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 	}
 }
 
+// updateNodeTaint updates node out-of-service taint
+func (az *Cloud) updateNodeTaint(node *v1.Node) {
+	if node == nil {
```

Comment:
`|| az.KubeClient == nil` could be merged together here.

Reply:
Updated. I made it separate to add detailed warning msg. Thanks